Add monitoring blocks to Identity Platform Config and Tenant resources

anuraaga opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->

Closes https://github.com/hashicorp/terraform-provider-google/issues/13326

This reintroduces #7025 - it is the exact same change with some modifications to apply to the current `main` branch (file was split into several). I was interested in managing identity platform with Terraform but found this missing config, and while it seemed there were some API issues when the original change was made, trying again now seems to be fine - the example tests pass on my local machine. Hopefully that means we can finally add this :)

```
❯ make testacc TEST=./google/services/identityplatform TESTARGS='-run=TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample -v'
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/identityplatform -v -run=TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample -v -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample
=== PAUSE TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample
=== CONT  TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample
--- PASS: TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample (76.88s)

❯ make testacc TEST=./google/services/identityplatform TESTARGS='-run=TestAccIdentityPlatformConfig_identityPlatformTenantMonitoringExample -v'
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/identityplatform -v -run=TestAccIdentityPlatformConfig_identityPlatformTenantMonitoringExample -v -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccIdentityPlatformConfig_identityPlatformTenantMonitoringExample
=== PAUSE TestAccIdentityPlatformConfig_identityPlatformTenantMonitoringExample
=== CONT  TestAccIdentityPlatformConfig_identityPlatformTenantMonitoringExample
--- PASS: TestAccIdentityPlatformConfig_identityPlatformTenantMonitoringExample (196.55s)
PASS
ok  	github.com/hashicorp/terraform-provider-google/google/services/identityplatform	197.171s
```

@bschaatsbergen I have added you as a co-author of the commit

<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
identity_platform: Added attribute block `monitoring` to `google_identity_platform_tenant`
```

```release-note:enhancement
identity_platform: Added attribute block `monitoring` to `google_identity_platform_config`
```

```release-note:enhancement
identity_platform: Added attribute block `multi_tenant_config` to `google_identity_platform_config`
```

modular-magician commented:
Hello! I am a robot. It looks like you are a: Community Contributor ~Googler~ ~Core Contributor~. Tests will require approval to run.

@shuyama1, a repository maintainer, has been assigned to review your changes. If you have not received review feedback within 2 business days, please leave a comment on this PR asking them to take a look.

You can help make sure that review is quick by [doing a self-review](https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/) and by [running impacted tests locally](https://googlecloudplatform.github.io/magic-modules/get-started/run-provider-tests/).

anuraaga commented:
It looks like there wasn't a CLA check in this repo when #7025 was introduced, @bschaatsbergen would you be able to sign the CLA? Or should I remove you from the commit? Let me know what works best

bschaatsbergen commented:
Hi @anuraaga I've definitely signed the CLA 🤔

anuraaga commented:
@bschaatsbergen Ah I see - in that case the email address I used in the commit may not be the one you used to sign the CLA (I took the same email as the one from your original commit). Can you share the email address or an example commit which would pass CLA check?

bschaatsbergen commented:
I usually commit under `git@bschaatsbergen.com` but it seems that my CLA is signed under 'schaatsbergenbruno@gmail.com' @anuraaga

anuraaga commented:
Thanks for confirming @bschaatsbergen - I have edited the commit with that address and all looks good now. Just as a heads up if you end up sending a PR in the future, you'll need to use that address, or I think there is a way to associate multiple addresses through the CLA tool but am not sure.

bschaatsbergen commented:
@slevenick could you take a look at this PR if possible? It includes a fix which we faced in #7025 but with a few slight changes.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8895-old..auto-pr-8895) ( 5 files changed, 493 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8895-old..auto-pr-8895) ( 5 files changed, 493 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8895-old..auto-pr-8895) ( 2 files changed, 136 insertions(+))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_identity_platform_config` (4 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_identity_platform_config" "primary" {
  multi_tenant {
    allow_tenants           = # value needed
    default_tenant_location = # value needed
  }
}

```
Resource: `google_identity_platform_tenant` (12 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_identity_platform_tenant" "primary" {
  monitoring {
    request_logging {
      enabled = # value needed
    }
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3022` 
Passed tests `2722` 
Skipped tests: `299` 
Affected tests: `1` 

#### Action taken 
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataprocClusterIamPolicy </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/e9febd72-b462-4e4b-88f0-e75506a13fa5/recording/TestAccDataprocClusterIamPolicy.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/e9febd72-b462-4e4b-88f0-e75506a13fa5/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/e9febd72-b462-4e4b-88f0-e75506a13fa5/recording) for each test

anuraaga commented:
Sorry for some confusion! I had some copy-paste problem so I wasn't actually running the test that was failing in #7025. I have updated to fix the tests and could confirm there was no backend change to magically have it start working.

So I tried debugging with curl requests and I found two fixes

-- Monitoring config on a tenant cannot be set at creation

I realized that the `POST` that creates a tenant has its monitoring config ignored. This seems to be a strict implementation of the docs

https://cloud.google.com/identity-platform/docs/activity-logging

> Activity logging for tenant projects is disabled by default. You can enable activity logging for tenant projects, using the REST API, by executing the following command:

It makes no mention of setting it while creating a tenant, presumably because it is not allowed by the backend. I do wish in that case it would reject the request with monitoring config instead of silently ignore :)

But to accomodate this, I updated the existing post_create script to send a PATCH after creation if monitoring config is set. I noticed `custom-code.md` specifically calls out this use case for post_create "Use `post_create` to set an update-only field after create finishes." so hopefully this is reasonable

-- Empty monitoring object in config only when multi-tenant is enabled

It seems that the initial config resource also always has an empty monitoring object returned when multi tenancy is enabled (possibly this is related to how backend implements the above "disabled by default" behavior). So I set `default_from_api: true` for `monitoring`. I'm not too sure if there are significant downsides to this, but it seems to be the only way to have the behavior match for single and multi-tenant.

Do these updates to the PR seem reasonable? Sorry again for some initial confusion.

shuyama1 commented:
@anuraaga Thanks for the explanation and apologize for the delay on review.  Adding `post_create` and `default_from_api` seems reasonable to resolve the issues occurred earlier. Would you mind rebasing your PR and resolve the merge conflicts, if any? Plus, any particular reasons we skip VCR tests for these tests? I see that you've added explicit dependency in each test and I'd assume that the tests should work well with VCR.

anuraaga commented on mmv1/products/identityplatform/Config.yaml:
```diff
@@ -73,10 +103,41 @@ properties:
     name: 'signIn'
     description: |
       Configuration related to local sign in methods.
+    default_from_api: true
```
I had to add to this field as well, it seems the API returns something even when not set

```
=== CONT  TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample
    vcr_utils.go:152: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # google_identity_platform_config.default will be updated in-place
          ~ resource "google_identity_platform_config" "default" {
                id                         = "projects/tf-testi2dmqeauyz/config"
                name                       = "projects/866775349519/config"
                # (3 unchanged attributes hidden)

              - sign_in {
                  - allow_duplicate_emails = false -> null
                  - hash_config            = [
                      - {
                          - algorithm      = "SCRYPT"
                          - memory_cost    = 14
                          - rounds         = 8
                          - salt_separator = "Bw=="
                          - signer_key     = "fxBwaqYujgOuuq3NG/aeDC/9x7c6oar9Hxyrr3GdzRuu+16oBveHuHjKiecAYNJFgHFz+091dPCwgrVnYoef6Q=="
                        },
                    ] -> null
                }

                # (1 unchanged block hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
```

shuyama1 replied:
Sorry, it seems like a bug introduced from a new feature last week. I already created a PR to fix it and hopefully the fix will be merged very soon. Thanks for catching it!

anuraaga replied:
Thanks @shuyama1 - I saw the fix is in and I have merged main into this PR

shuyama1 replied:
Thanks!

anuraaga reviewed (commented):
> Plus, any particular reasons we skip VCR tests for these tests? I see that you've added explicit dependency in each test and I'd assume that the tests should work well with VCR.

Sorry I had just copied from the other test but it seems fine without it

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8895-old..auto-pr-8895) ( 5 files changed, 598 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8895-old..auto-pr-8895) ( 5 files changed, 598 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8895-old..auto-pr-8895) ( 2 files changed, 136 insertions(+))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_identity_platform_config` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_identity_platform_config" "primary" {
  multi_tenant {
    default_tenant_location = # value needed
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3072` 
Passed tests `2761` 
Skipped tests: `299` 
Affected tests: `12` 

#### Action taken 
<details> <summary>Found 12 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataprocClusterIamPolicy|TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample|TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantExample|TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantMonitoringExample|TestAccIdentityPlatformTenant_identityPlatformTenantBasicExample|TestAccIdentityPlatformTenant_identityPlatformTenantUpdate|TestAccIdentityPlatformTenantOauthIdpConfig_identityPlatformTenantOauthIdpConfigUpdate|TestAccIdentityPlatformTenantDefaultSupportedIdpConfig_identityPlatformTenantDefaultSupportedIdpConfigBasicExample|TestAccIdentityPlatformTenantInboundSamlConfig_identityPlatformTenantInboundSamlConfigUpdate|TestAccIdentityPlatformTenantOauthIdpConfig_identityPlatformTenantOauthIdpConfigBasicExample|TestAccIdentityPlatformTenantInboundSamlConfig_identityPlatformTenantInboundSamlConfigBasicExample|TestAccIdentityPlatformTenantDefaultSupportedIdpConfig_identityPlatformTenantDefaultSupportedIdpConfigUpdate </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccDataprocClusterIamPolicy.log)]
`TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformConfig_identityPlatformConfigMonitoringExample.log)]
`TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantExample.log)]
`TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantMonitoringExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantMonitoringExample.log)]
`TestAccIdentityPlatformTenant_identityPlatformTenantBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenant_identityPlatformTenantBasicExample.log)]
`TestAccIdentityPlatformTenant_identityPlatformTenantUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenant_identityPlatformTenantUpdate.log)]
`TestAccIdentityPlatformTenantOauthIdpConfig_identityPlatformTenantOauthIdpConfigUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenantOauthIdpConfig_identityPlatformTenantOauthIdpConfigUpdate.log)]
`TestAccIdentityPlatformTenantDefaultSupportedIdpConfig_identityPlatformTenantDefaultSupportedIdpConfigBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenantDefaultSupportedIdpConfig_identityPlatformTenantDefaultSupportedIdpConfigBasicExample.log)]
`TestAccIdentityPlatformTenantInboundSamlConfig_identityPlatformTenantInboundSamlConfigUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenantInboundSamlConfig_identityPlatformTenantInboundSamlConfigUpdate.log)]
`TestAccIdentityPlatformTenantOauthIdpConfig_identityPlatformTenantOauthIdpConfigBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenantOauthIdpConfig_identityPlatformTenantOauthIdpConfigBasicExample.log)]
`TestAccIdentityPlatformTenantInboundSamlConfig_identityPlatformTenantInboundSamlConfigBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenantInboundSamlConfig_identityPlatformTenantInboundSamlConfigBasicExample.log)]
`TestAccIdentityPlatformTenantDefaultSupportedIdpConfig_identityPlatformTenantDefaultSupportedIdpConfigUpdate`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording/TestAccIdentityPlatformTenantDefaultSupportedIdpConfig_identityPlatformTenantDefaultSupportedIdpConfigUpdate.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/90d811b9-060a-4032-af43-69d66a5a835e/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8895-old..auto-pr-8895) ( 5 files changed, 598 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8895-old..auto-pr-8895) ( 5 files changed, 598 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8895-old..auto-pr-8895) ( 2 files changed, 136 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `3078` 
Passed tests `2775` 
Skipped tests: `299` 
Affected tests: `4` 

#### Action taken 
<details> <summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample|TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample|TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample|TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/56038bd7-a737-4b05-a882-ea3b17d58fb1/recording/TestAccBigtableAppProfile_bigtableAppProfileAnyclusterExample.log)]
`TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/56038bd7-a737-4b05-a882-ea3b17d58fb1/recording/TestAccBigtableAppProfile_bigtableAppProfileSingleclusterExample.log)]
`TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/56038bd7-a737-4b05-a882-ea3b17d58fb1/recording/TestAccBigtableAppProfile_bigtableAppProfileMulticlusterExample.log)]
`TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/56038bd7-a737-4b05-a882-ea3b17d58fb1/recording/TestAccIdentityPlatformConfig_identityPlatformConfigMultiTenantExample.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/56038bd7-a737-4b05-a882-ea3b17d58fb1/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-8895/artifacts/56038bd7-a737-4b05-a882-ea3b17d58fb1/recording) for each test

shuyama1 commented on mmv1/products/identityplatform/Tenant.yaml:
```diff
@@ -52,6 +52,20 @@ properties:
+    description: |
+      Configuration related to monitoring project activity.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
```
Same here for making subfields required

shuyama1 commented on mmv1/products/identityplatform/Config.yaml:
```diff
@@ -72,6 +102,36 @@ properties:
+    description: |
+      Configuration related to multi-tenant functionality.
+    properties:
+      - !ruby/object:Api::Type::Boolean
```
Can we have some test coverage on negative value for boolean fields. Just want to make sure the API returns the value back if we explicitly set the `false` value, otherwise, it will cause a permadiff and we may need to add a custom flattener to handle it

anuraaga commented:
I'm really sorry - I had some time traveling since the last commit and haven't been able to get back to this, mostly because the test cycle in this repo is very hard and actually costs some money. I hope someone will be able to take over this change as unfortunately I won't be able to.
